meta-facebook/gt-cc/src/platform/plat_isr.c

```diff
@@ -106,3 +103,157 @@ void ISR_DC_ON()
+				       PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL;
+	event.previous_event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL :
+						PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT;
+
+	LOG_WRN("PEX ADC is %s", is_alert ? "alert" : "non-alert");
+
+	if (pldm_send_platform_event(PLDM_SENSOR_EVENT, PLDM_EVENT_SENSOR_PEX,
+				     PLDM_STATE_SENSOR_STATE, (uint8_t *)&event,
+				     sizeof(struct pldm_sensor_event_state_sensor_state))) {
+		LOG_ERR("Send PEX ADC alert event log failed");
+	}
+}
+
+void ISR_SMB_FPGA_ALERT()
+{
+	struct pldm_sensor_event_state_sensor_state event;
+
+	bool is_alert = !gpio_get(SMB_FPGA_ALERT_R_N);
+
+	event.sensor_offset = PLDM_STATE_SET_OFFSET_DEVICE_STATUS;
+	event.event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT :
+				       PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL;
+	event.previous_event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL :
+						PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT;
+
+	LOG_WRN("FPGA SMB is %s", is_alert ? "alert" : "non-alert");
+
+	if (pldm_send_platform_event(PLDM_SENSOR_EVENT, PLDM_EVENT_SENSOR_CPLD,
+				     PLDM_STATE_SENSOR_STATE, (uint8_t *)&event,
+				     sizeof(struct pldm_sensor_event_state_sensor_state))) {
+		LOG_ERR("Send FPGA SMB alert event log failed");
+	}
+}
+
+void ISR_VR_PMBUS_ALERT()
+{
+	struct pldm_sensor_event_state_sensor_state event;
+
+	bool is_alert = !gpio_get(SMB_ALERT_PMBUS_R_N);
+
+	event.sensor_offset = PLDM_STATE_SET_OFFSET_DEVICE_STATUS;
+	event.event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT :
+				       PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL;
+	event.previous_event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL :
+						PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT;
+
+	LOG_WRN("VR PMBUS is %s", is_alert ? "alert" : "non-alert");
+
+	if (pldm_send_platform_event(PLDM_SENSOR_EVENT, PLDM_EVENT_SENSOR_VR,
+				     PLDM_STATE_SENSOR_STATE, (uint8_t *)&event,
+				     sizeof(struct pldm_sensor_event_state_sensor_state))) {
+		LOG_ERR("Send VR PMBUS alert event log failed");
+	}
+}
+
+void ISR_HSC_SMB_ALERT()
+{
+	struct pldm_sensor_event_state_sensor_state event;
+
+	bool is_alert = !gpio_get(SMB_ALERT_HSC_R_N);
+
+	event.sensor_offset = PLDM_STATE_SET_OFFSET_DEVICE_STATUS;
+	event.event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT :
+				       PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL;
+	event.previous_event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL :
+						PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT;
+
+	LOG_WRN("HSC SMB is %s", is_alert ? "alert" : "non-alert");
+
+	if (pldm_send_platform_event(PLDM_SENSOR_EVENT, PLDM_EVENT_SENSOR_HSC,
+				     PLDM_STATE_SENSOR_STATE, (uint8_t *)&event,
+				     sizeof(struct pldm_sensor_event_state_sensor_state))) {
+		LOG_ERR("Send HSC SMB alert event log failed");
+	}
+}
+
+#define ISR_SSD_PRESENT_HANDLER(idx)                                                               \
```

Comment:
Is there a reason for all the line continuation characters in this function?

Reply:
Hi @garnermic,

We use all the line continuation to define a multiple-line macro as a function those ones are the same but differ by the index for the SSD0 - SSD15. Is there anything risk we haven't considered? We can expand all those functions to normal function declaration and definition.

Thanks.